File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @param {number} k
     * @return {number}
     */
    characterReplacement(s, k) {
        const map={};
         let left = 0 ;
        let right = 0 ;
        let mostFreqCharCount = 0 ;
   
    for(right = 0 ; right < s.length ; right++){
        map[s[right]]=(map[s[right]] || 0)+1;
        mostFreqCharCount = Math.max(mostFreqCharCount, map[s[right]]);
        if(right - left + 1   > k+mostFreqCharCount ){
            map[s[left++]]-- ;
        }
    }
    return right - left ;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
  int c=0;

    void dfs(TreeNode *root,int m){
        if(!root) return;
        if(root->val >= m) c++;
        m=max(m,root->val);
        dfs(root->left,m);
        dfs(root->right,m);
        m=max(m,root->val);

    }
public:
    int goodNodes(TreeNode* root) {
        dfs(root,-100);
        return c;
    }
};
